Index: mozilla/nsprpub/pr/include/prlog.h
===================================================================
RCS file: /cvsroot/mozilla/nsprpub/pr/include/prlog.h,v
retrieving revision 3.16
diff -p -u -8 -r3.16 prlog.h
--- mozilla/nsprpub/pr/include/prlog.h	22 Feb 2009 19:56:13 -0000	3.16
+++ mozilla/nsprpub/pr/include/prlog.h	15 Jan 2010 01:33:24 -0000
@@ -184,16 +184,18 @@ NSPR_API(void) PR_SetLogBuffering(PRIntn
 */
 NSPR_API(void) PR_LogPrint(const char *fmt, ...);
 
 /*
 ** Flush the log to its file.
 */
 NSPR_API(void) PR_LogFlush(void);
 
+NSPR_API(void) PR_Assert(const char *s, const char *file, PRIntn ln);
+
 #if defined(DEBUG) || defined(FORCE_PR_LOG)
 #define PR_LOGGING 1
 
 #define PR_LOG_TEST(_module,_level) \
     ((_module)->level >= (_level))
 
 /*
 ** Log something.
@@ -228,17 +230,16 @@ NSPR_API(void) PR_LogFlush(void);
 #define PR_LOG_END(module,level,args)
 #define PR_LOG_DEFINE(_name)    NULL
 #endif /* PR_LOGGING */
 
 #endif /* NO_NSPR_10_SUPPORT */
 
 #if defined(DEBUG) || defined(FORCE_PR_ASSERT)
 
-NSPR_API(void) PR_Assert(const char *s, const char *file, PRIntn ln);
 #define PR_ASSERT(_expr) \
     ((_expr)?((void)0):PR_Assert(# _expr,__FILE__,__LINE__))
 
 #define PR_NOT_REACHED(_reasonStr) \
     PR_Assert(_reasonStr,__FILE__,__LINE__)
 
 #else
 
Index: mozilla/nsprpub/pr/src/md/windows/w95sock.c
===================================================================
RCS file: /cvsroot/mozilla/nsprpub/pr/src/md/windows/w95sock.c,v
retrieving revision 3.16
diff -p -u -8 -r3.16 w95sock.c
--- mozilla/nsprpub/pr/src/md/windows/w95sock.c	4 Feb 2009 23:44:01 -0000	3.16
+++ mozilla/nsprpub/pr/src/md/windows/w95sock.c	15 Jan 2010 01:33:24 -0000
@@ -67,17 +67,17 @@ static PRInt32 socket_io_wait(
 
 typedef enum _WSA_COMPATIBILITY_BEHAVIOR_ID {
     WsaBehaviorAll = 0,
     WsaBehaviorReceiveBuffering,
     WsaBehaviorAutoTuning
 } WSA_COMPATIBILITY_BEHAVIOR_ID, *PWSA_COMPATIBILITY_BEHAVIOR_ID;
 
 /* from sdkddkver.h */
-#define NTDDI_LONGHORN                      0x06000000
+#define NTDDI_WIN6              0x06000000  /* Windows Vista */
 
 /* from winsock2.h */
 #define WSAEVENT                HANDLE
 
 #define WSAOVERLAPPED           OVERLAPPED
 typedef struct _OVERLAPPED *    LPWSAOVERLAPPED;
 
 typedef void (CALLBACK * LPWSAOVERLAPPED_COMPLETION_ROUTINE)(
@@ -170,17 +170,17 @@ _PR_MD_SOCKET(int af, int type, int flag
     if ((af == AF_INET || af == AF_INET6) && 
         type == SOCK_STREAM && socketSetCompatMode)
     {
         WSA_COMPATIBILITY_MODE mode;
         char dummy[4];
         int ret_dummy;
 
         mode.BehaviorId = WsaBehaviorAutoTuning;
-        mode.TargetOsVersion = NTDDI_LONGHORN;
+        mode.TargetOsVersion = NTDDI_WIN6;
         if (wsaioctlProc(sock, SIO_SET_COMPATIBILITY_MODE,  
                          (char *)&mode, sizeof(mode),
                          dummy, 4, &ret_dummy, 0, NULL) == SOCKET_ERROR)
         {
             int err = WSAGetLastError();
             PR_LOG(_pr_io_lm, PR_LOG_DEBUG, ("WSAIoctl() failed with %d", err));
 
             /* SIO_SET_COMPATIBILITY_MODE may not be supported.
